### DISCUSSION TITLE
How to compile and run my source code？

### DISCUSSION THREAD

**Discussion by biaogxb** [idaholab/moose#20431] [Q&A Getting Started] [answered]
I just started learning to use moose, and my source code: .c documents and .h and .i file, how to calculate on my moose and get the result？Do I need to write the makefile by myself?

**Comment by loganharbour** [comment, marked as answer]
Follow the instructions here to create an application: https://mooseframework.inl.gov/getting_started/new_users.html

After the application is made, you can place your source in `src` and `include` and the pre-generated makefile will compile appropriately.

**Reply by biaogxb** [reply]
Thank you for your answer. I tried this, After compiling, I tried to run input.i file in my registered application, but this error occurred. Do you know how to overcome this problem.
the following errors were reported：
A 'Migration' is not a registered object.

If you are trying to find this object in a dynamically linked library, make sure that
the library can be found either in your "Problem/library_path" parameter or in the
MOOSE_LIBRARY_PATH environment variable.
[Code.zip](https://github.com/idaholab/moose/files/8162885/Code.zip)

**Reply by GiudGiud** [reply]
See https://github.com/idaholab/moose/discussions/20436

**Reply by biaogxb** [reply]
thank you so much！！!
